kernel/sucompat.c

```diff
@@ -99,25 +99,25 @@ int ksu_handle_execveat_sucompat(int *fd, struct filename **filename_ptr,
 	const char sh[] = KSUD_PATH;
 	const char su[] = SU_PATH;
 
-	if (!filename_ptr)
+	if (unlikely(!filename_ptr))
 		return 0;
 
 	filename = *filename_ptr;
 	if (IS_ERR(filename)) {
 		return 0;
 	}
 
-	if (!ksu_is_allow_uid(current_uid().val)) {
-		return 0;
-	}
-
-	if (!memcmp(filename->name, su, sizeof(su))) {
+	if (unlikely(!memcmp(filename->name, su, sizeof(su)))) {
```

Comment:
1. Perhaps we should add some comments here, as it might give the impression that any app can acquire root just by invoking su; however, in reality, apps that are not allowed root access will be denied at the stat and faccessat stages.
2. Would the existence of /system/bin/su in the system lead to root authorization being invalidated, allowing any app to unconditionally obtain root?

Reply:
Ah, didn't review the logic inside the memcmp() check carefully, sorry for that.

We can just return early if memcmp() check fails. I'll reorder the code so that the behavior is consistent with the older one.